WooCommerce/Classes/ViewRelated/Orders/Order Details/AddOns/OrderAddOnTopBanner.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+import SwiftUI
+
+struct OrderAddOnTopBanner: UIViewRepresentable {
+    typealias Callback = () -> ()
+
+    /// Desired `width` of the view.
+    ///
+    private let width: CGFloat
+
+    /// Closure to be invoked when the "Give Feedback" button is pressed.
+    ///
+    private var onGiveFeedback: Callback? = nil
+
+    /// Closure to be invoked when the "Dismiss" button is pressed.
+    ///
+    private var onDismiss: Callback? = nil
```

Comment:
I am wondering why these are optional instead of part of the initializer. If the view is always going to show these two buttons, should it ensure there is a callback to handle them?

Reply:
Just for syntactic sugar.

After looking at some native `SwiftUI` components, I decided to add the closure as a modifier rather than as a parameter as it reads a little bit better.

```swift
var body: some view {
    OrderAddOnTopBanner(onGiveFeedback: {
         // Give Feedback
    }, onDismiss: {
         // Dismiss
    })
        .bodyStyle()
	    .fixedSize()
}
```

VS

```swift
var body: some view {
    OrderAddOnTopBanner(width: geometry.size.width)
        .onDismiss {
            //dismiss
        }
        .onGiveFeedback {
            // Give Feedback
        }
        .bodyStyle()
        .fixedSize()
}
```

I could go either way if you feel strongly about it tho!